pkgs/data/misc/cacert/default.nix

```
@@ -31,93 +22,160 @@ stdenv.mkDerivation rec {
     sha256 = "1jfdnh5l4k57r2vb07s06hqi7m2qzk0d9x25lsdsrw3cflx9x9w9";
   };
 
-  outputs = [ "out" "unbundled" ];
+  outputs = [ "out" "unbundled" "p11kit" ];
 
-  nativeBuildInputs = [ python3 ];
+  nativeBuildInputs = [ buildcatrust ];
 
   configurePhase = ''
     ln -s nss/lib/ckfw/builtins/certdata.txt
 
-    cat << EOF > blacklist.txt
-    ${lib.concatStringsSep "\n" (map (c: ''"${c}"'') blacklist)}
+    cat << EOF > blocklist.txt
+    ${lib.concatStringsSep "\n" blacklist}
     EOF
 
-    # copy from the store, otherwise python will scan it for imports
-    cat "${certdata2pem}" > certdata2pem.py
+    cat << EOF > extra-certificates-bundle.crt
+    ${lib.concatStringsSep "\n\n" extraCertificateStrings}
+    EOF
   '';
 
   buildPhase = ''
-    python certdata2pem.py | grep -vE '^(!|UNTRUSTED)'
-
-    for cert in *.crt; do
-      echo $cert | cut -d. -f1 | sed -e 's,_, ,g' >> ca-bundle.crt
-      cat $cert >> ca-bundle.crt
-      echo >> ca-bundle.crt
-    done
+    mkdir unbundled
+    buildcatrust \
+      --certdata_input certdata.txt \
+      --ca_bundle_input extra-certificates-bundle.crt ${lib.concatStringsSep " " extraCertificateFiles} \
+      --blocklist blocklist.txt \
+      --ca_bundle_output ca-bundle.crt \
+      --ca_unpacked_output unbundled \
+      --p11kit_output ca-bundle.trust.p11-kit
   '';
 
   installPhase = ''
-    mkdir -pv $out/etc/ssl/certs
-    cp -v ca-bundle.crt $out/etc/ssl/certs
+    install -D ca-bundle.crt $out/etc/ssl/certs/ca-bundle.crt
```

```suggestion
    install -D ca-bundle.crt "$out/etc/ssl/certs/"
```
Variable should always be quoted, better safe than sorry.